membership_extension/models/product_template.py

```diff
@@ -19,3 +22,11 @@ def _get_next_date(self, date, qty=1):
             date_to = fields.Date.from_string(self.membership_date_to)
             return date_to + timedelta(1)
         return False  # pragma: no cover
+
+    @api.onchange('company_id')
+    def _onchange_company_id_compute_membership_category_id(self):
+        """Reset the Membership Category in case a different Company is set"""
+        category_company = self.membership_category_id.company_id
+        if self.company_id and category_company:
+            if not category_company == self.company_id:
```

Comment:
Weird way to put it...
```suggestion
            if category_company != self.company_id:
```